### PR TITLE
Add methods for adding test sources & resources in code generators

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/35933

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java [hunk 1]
```diff
@@ -297,6 +304,30 @@ public void addSourceFile(TextDocument textDocument, String filenamePrefix) {
             addSourceFile(textDocument, filenamePrefix, defaultModuleId);
         }
 
+        @Override
+        public void addTestFile(TextDocument textDocument, String filenamePrefix, ModuleId moduleId) {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java:308]: Let's rename this to `addTestSourceFile`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/SourceGeneratorContext.java [hunk 2]
```diff
@@ -61,6 +61,23 @@ public interface SourceGeneratorContext {
...
     void addSourceFile(TextDocument textDocument, String filenamePrefix);
 
+    /**
+     * Adds the provided test source code as a new file to the {@code Module} identified by the {@code ModuleId}.
+     *
+     * @param textDocument   a {@code TextDocument} that contains the test source code
+     * @param filenamePrefix proposed prefix of the filename to be added
+     * @param moduleId       indicates the module to which the new test source file should be added
+     */
+    void addTestFile(TextDocument textDocument, String filenamePrefix, ModuleId moduleId);
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/SourceGeneratorContext.java:71]: Refer https://github.com/ballerina-platform/ballerina-lang/pull/35934/files#r859531284

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java]: Renamed via https://github.com/ballerina-platform/ballerina-lang/pull/35934/commits/f00062fd30e7ddbad3bcc66aeee8c2ec2b81275d

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/SourceGeneratorContext.java (hunk 2 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/SourceGeneratorContext.java]: Renamed via https://github.com/ballerina-platform/ballerina-lang/pull/35934/commits/f00062fd30e7ddbad3bcc66aeee8c2ec2b81275d

**Review by azinneera — APPROVED** [review]

**Pull request merged**